dot/parachain/prospective-parachains/prospective_parachains_test.go

```diff
@@ -0,0 +1,84 @@
+package prospectiveparachains
+
+import (
+	"testing"
+
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockFragmentChain struct {
+	fragmentChain
+}
+
+func (m *MockFragmentChain) FindBackableChain(
+	ancestors Ancestors,
+	qty uint32,
+) []parachaintypes.CandidateHashAndRelayParent {
+	return []parachaintypes.CandidateHashAndRelayParent{
+		{
+			CandidateHash: parachaintypes.CandidateHash{
+				Value: common.Hash{0x10},
+			},
+			CandidateRelayParent: common.Hash{0x20},
+		},
+		{
+			CandidateHash: parachaintypes.CandidateHash{
+				Value: common.Hash{0x11},
+			},
+			CandidateRelayParent: common.Hash{0x21},
+		},
+	}
+}
+
+func TestGetBackableCandidates(t *testing.T) {
```

Comment:
Please add more tests to cover the various cases that result in an empty response.

You probably want to use a generated mock for that. [Here](https://github.com/ChainSafe/gossamer/blob/feat/parachain/dot/parachain/backing/mocks_generate_test.go) is an example for how to do that.

Reply:
@haikoschol i trying to refactor these tests and i will let you know